projects/optic/src/commands/diff/diff.ts

```diff
@@ -232,22 +245,57 @@ const getDiffAction =
     const diffResult = await runDiff(files, parsedFiles, config, options);
     if (config.isAuthenticated) {
       const [baseParseResult, headParseResult] = parsedFiles;
-      const apiId: string | null = 'TODO'; // headParseResult.jsonLike[OPTIC_URL_KEY] ?? baseParseResult.jsonLike[OPTIC_URL_KEY] ?? null
-      const shouldUploadBaseSpec = baseParseResult.context && apiId;
-      const shouldUploadHeadSpec = headParseResult.context && apiId;
-      if (shouldUploadBaseSpec) {
-        // TODO upload spec
+      const opticUrl: string | null =
+        headParseResult.jsonLike[OPTIC_URL_KEY] ??
+        baseParseResult.jsonLike[OPTIC_URL_KEY] ??
+        null;
+      const specDetails = opticUrl ? getApiFromOpticUrl(opticUrl) : null;
+      // We upload a spec if it is unchanged in git and there is an API id on the spec
+      let baseSpecId: string | null = null;
+      let headSpecId: string | null = null;
+      if (baseParseResult.context && specDetails) {
+        const tags =
+          baseParseResult.context.vcs === VCS.Git
+            ? [`git:${baseParseResult.context.sha}`]
+            : [];
+        baseSpecId = await uploadSpec(specDetails.apiId, {
+          spec: baseParseResult,
+          client: config.client,
+          tags,
+        });
+      } else if (baseParseResult.isEmptySpec) {
+        baseSpecId = EMPTY_SPEC_ID;
       }
-      if (shouldUploadHeadSpec) {
-        // TODO upload spec
+      if (headParseResult.context && specDetails) {
+        let tags: string[] = [];
+        if (headParseResult.context.vcs === VCS.Git) {
+          const currentBranch = await Git.getCurrentBranchName();
+          tags = [`git:${headParseResult.context.sha}`, `git:${currentBranch}`];
```

Comment:
I would change the branch tag prefix to be something like `branch` or `gitbranch`